File: public/java/decryptDataInKms.java
//
// Decrypt data in KMS
//
public static void decryptDataInKms(String kmsServerUrl, String apiKey, String userDecryptionKey, String ciphertext) {
  KmsClient kmsClient = new KmsClient(kmsServerUrl, apiKey);
  String plaintext = new String(
    kmsClient.coverCryptDecrypt(userDecryptionKey, ciphertext)
    .getPlaintext(),
  StandardCharsets.UTF_8);
}

File: public/javascript/decryptDataInKms.ts
import { KmsClient } from "cloudproof_js";
import { BACKEND_URL } from "./backendConfig";

//
// Decrypt data in KMS
//
export const decryptDataInKms = async (encryptText: Uint8Array, kmsToken: string, decryptionKeyUid: string): Promise<string> => {
  const client = new KmsClient(BACKEND_URL, kmsToken);
  const decryptedData = await client.coverCryptDecrypt(decryptionKeyUid, encryptText);
  return new TextDecoder().decode(decryptedData.plaintext);
};

export const bulkDecryptInKms = async (encryptData: Uint8Array[], kmsToken: string, decryptionKey: string): Promise<string[]> => {
  const client = new KmsClient(BACKEND_URL, kmsToken);
  const decryptedData = await client.coverCryptBulkDecrypt(decryptionKey, encryptData);
  return decryptedData.plaintext.map((bits) => new TextDecoder().decode(bits));
};
